File: src/main/java/com/soapdogg/lispInterpreter/generator/NodeGenerator.java
package com.soapdogg.lispInterpreter.generator;

import com.soapdogg.lispInterpreter.datamodels.AtomNode;
import com.soapdogg.lispInterpreter.datamodels.ExpressionNode;
import com.soapdogg.lispInterpreter.datamodels.Node;
import com.soapdogg.lispInterpreter.constants.ReservedValuesConstants;
import lombok.AllArgsConstructor;

@AllArgsConstructor(staticName = "newInstance")
public class NodeGenerator {

    public AtomNode generateAtomNode(
        final boolean value
    ) {
        String t = value ? ReservedValuesConstants.T : ReservedValuesConstants.NIL;
        return new AtomNode(
            t
        );
    }

    public AtomNode generateAtomNode(
        final int value
    ) {
        String t = Integer.toString(value);
        return new AtomNode(
            t
        );
    }

    public AtomNode generateAtomNode(
        final String value
    ) {
        return new AtomNode(value);
    }

    public ExpressionNode generateExpressionNode(
        final Node address,
        final Node data
    ) {
        return new ExpressionNode(
            address,
            data
        );
    }
}

File: src/main/java/com/soapdogg/lispInterpreter/generator/NodeGenerator.kt
package com.soapdogg.lispInterpreter.generator

import com.soapdogg.lispInterpreter.constants.ReservedValuesConstants
import com.soapdogg.lispInterpreter.datamodels.*

class NodeGenerator {
    fun generateAtomNode(
        value: Boolean
    ): AtomNode {
        val t = if (value) ReservedValuesConstants.T else ReservedValuesConstants.NIL
        return AtomNode(
            t
        )
    }

    fun generateAtomNode(
        value: Int
    ): AtomNode {
        val t = value.toString()
        return AtomNode(
            t
        )
    }

    fun generateAtomNode(
        value: String
    ): AtomNode {
        return AtomNode(value)
    }

    fun generateExpressionListNode(
        children: List<NodeV2>
    ): ExpressionListNode {
        return ExpressionListNode(
            children
        )
    }
}